src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -74,7 +72,11 @@ internal sealed class TracerProviderSdk : TracerProvider
             }
         }
 
-        foreach (var processor in state.Processors)
+        var processors = (IEnumerable<BaseProcessor<Activity>>)state.Processors.OrderBy(p => (int)p.PipelineWeight);
```

Comment:
`PipelineWeight` is an `int`.

```suggestion
        var processors = (IEnumerable<BaseProcessor<Activity>>)state.Processors.OrderBy(p => p.PipelineWeight);
```